### PR TITLE
Start using Minio::Client instead of AWS SDK for PG backup operations

### PR DESCRIPTION
**Pull request by furkansahin**
### Change return type of list_objects for non existent bucket
When buckets do not exist, we need to gracefully return empty list.
Otherwise, this causes a failure in the postgres service provisioning
path. The main reason is, at the time of creation of the object, the
bucket for backup does not exist, yet. However, the moment we are
redirected to the postgres object page, we send a query to the MinIO
service. Just returning empty array is more than enough to fix that.
### Remove AWS S3 SDK related code

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,6 @@ source "https://rubygems.org"
 ruby "3.2.2"
 
 gem "argon2"
-gem "aws-sdk-s3"
```
Comment (enescakir) [inline Gemfile:7]: Thanks to god. They release 4-5 new versions every week.

File: model/postgres/postgres_timeline.rb [hunk 2]
```diff
@@ -47,13 +47,13 @@ def need_backup?
 
   def backups
     blob_storage_client
-      .list_objects(bucket_name: ubid, folder_path: "basebackups_005/")
-      .select { _1.key.end_with?("backup_stop_sentinel.json") }
+      .list_objects(ubid, "basebackups_005/")
+      .select { _1.object_name.end_with?("backup_stop_sentinel.json") }
```
Comment (enescakir) [inline model/postgres/postgres_timeline.rb:51]: Should we keep the parameters/properties of our minio client compatible  with the AWS S3 ruby client?

Comment (furkansahin) [reply model/postgres/postgres_timeline.rb]: solved it

**Review by byucesoy — APPROVED** [review]
I tested this E2E with PostgreSQL backup/restore. It all works nicely. :+1:

**Review by byucesoy — APPROVED** [review]

**Pull request merged**